zipline/sources/data_frame_source.py

```diff
@@ -138,6 +145,12 @@ def raw_data_gen(self):
             df = self.data.major_xs(dt)
             for sid, series in df.iteritems():
                 if sid in self.sids:
+                    # Skip SIDs that can not be forward filled
```

Comment:
Do you think there is a way to merge this logic with the dataframesource? They seem really close.